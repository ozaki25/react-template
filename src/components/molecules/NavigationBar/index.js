import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import NavigationBarLink from 'src/components/atoms/NavigationBarLink';
import NavigationTitleText from 'src/components/atoms/NavigationTitleText';

function NavigationBar({ title, onClickBack, onClickForward }) {
  return (
    <Container>
      <Left>
        {onClickBack && <NavigationBarLink onClick={onClickBack}>&lt;</NavigationBarLink>}
      </Left>
      <Title>
        <NavigationTitleText>{title}</NavigationTitleText>
      </Title>
      <Right>
        {onClickForward && <NavigationBarLink onClick={onClickForward}>&gt;</NavigationBarLink>}
      </Right>
    </Container>
  );
}

export const Container = styled.nav`
  border-bottom: 3px solid #e70000;
  height: 56px;
  text-align: center;
  width: '100%';
`;

export const Left = styled.div`
  float: left;
  padding-top: 10px;
  width: 50px;
`;

export const Title = styled.div`
  float: left;
  padding-top: 15px;
  width: calc(100% - ${50 * 2}px);
`;

export const Right = styled.div`
  float: left;
  padding-top: 10px;
  width: 50px;
`;

NavigationBar.propTypes = {
  title: PropTypes.string.isRequired,
  onClickBack: PropTypes.func,
  onClickForward: PropTypes.func,
};

NavigationBar.defaultProps = {
  onClickBack: null,
  onClickForward: null,
};

export default NavigationBar;
