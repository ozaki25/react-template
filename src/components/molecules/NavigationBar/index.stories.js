import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import NavigationBar from '.';

const stories = storiesOf('molecules', module);

stories
  .add(
    'NavigationBar/all props',
    withInfo('NavigationBar')(() => (
      <NavigationBar
        title={text('title', 'タイトル')}
        onClickBack={action('Back')}
        onClickForward={action('Forward')}
      />
    )),
  )
  .add(
    'NavigationBar/minimum props',
    withInfo('NavigationBar')(() => <NavigationBar title={text('title', 'タイトル')} />),
  );
