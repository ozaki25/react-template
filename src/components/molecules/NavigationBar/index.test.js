import React from 'react';
import { snapshot } from 'test/helpers';
import NavigationBar from '.';

const props = {
  title: 'タイトル',
  onClickBack: jest.fn(),
  onClickForward: jest.fn(),
};

snapshot('NavigationBar', <NavigationBar {...props} />);
snapshot('NavigationBar', <NavigationBar title="タイトル" />);
