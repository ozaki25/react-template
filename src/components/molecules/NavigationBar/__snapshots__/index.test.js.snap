// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: NavigationBar 1`] = `
<styled.nav>
  <styled.div>
    <NavigationBarLink
      onClick={[Function]}
    >
      &lt;
    </NavigationBarLink>
  </styled.div>
  <styled.div>
    <NavigationTitleText>
      タイトル
    </NavigationTitleText>
  </styled.div>
  <styled.div>
    <NavigationBarLink
      onClick={[Function]}
    >
      &gt;
    </NavigationBarLink>
  </styled.div>
</styled.nav>
`;

exports[`should match snapshot: NavigationBar 2`] = `
<styled.nav>
  <styled.div />
  <styled.div>
    <NavigationTitleText>
      タイトル
    </NavigationTitleText>
  </styled.div>
  <styled.div />
</styled.nav>
`;

exports[`should match snapshot: NavigationBar/Container 1`] = `
.c0 {
  border-bottom: 3px solid #e70000;
  height: 56px;
  text-align: center;
  width: '100%';
}

<nav
  className="c0"
/>
`;

exports[`should match snapshot: NavigationBar/Left 1`] = `
.c0 {
  float: left;
  padding-top: 10px;
  width: 50px;
}

<div
  className="c0"
/>
`;

exports[`should match snapshot: NavigationBar/Right 1`] = `
.c0 {
  float: left;
  padding-top: 10px;
  width: 50px;
}

<div
  className="c0"
/>
`;

exports[`should match snapshot: NavigationBar/Title 1`] = `
.c0 {
  float: left;
  padding-top: 15px;
  width: calc(100% - 100px);
}

<div
  className="c0"
/>
`;
